fastq_filtrator.py

```diff
@@ -0,0 +1,103 @@
+## Functions
+# This is a function that returns "Pass" if gc content of a read is between gc_bounds and "Fail" if it's not
+def gc_filter(seq, gc_bounds):
+    if len(gc_bounds) == 2:
+        min_gc, max_gc = int(gc_bounds[0]), int(gc_bounds[1])
+    elif len(gc_bounds) == 1:
+        min_gc, max_gc = int(gc_bounds[0]), 0 
+    else:
+        min_gc, max_gc = 0, 100
+    gc_count = int((seq.count('G') + seq.count('C')) * 100 / len(seq))
+    if ((gc_count < max_gc) and (gc_count > min_gc)):
+        return "Pass"
+    else:
+        return "Fail"
+ 
+# This is a function that counts an average qual phred33 for a read and returns "Pass"  if qual is more than a quality threshhold or "Fail" if it's not
+# 1. creating Qlist
+a = []
+for number in range(0,41):
+    a += [number]
+# 2. decoder function
+def decode_qual(qual_seq, *quality_threshold):
+    if len(quality_threshold) == 1:
+        min_qual = int(quality_threshold[0])
+    else:
+        min_qual = 0
+    q_quant = 0
+    for q in qual_seq:
+        q_quant += a[ord(q)-33]
+    qual_count = int(q_quant/len(qual_seq))
+    if (qual_count >= min_qual):
+        return "Pass"
+    else:
+        return "Fail"
+
+# This is a function that counts the len of a read and returns "Pass" (if it's more than a min_len and less than a max_len) or "Fail" if it's not
+def len_filter(read, length_bounds):
+    if len(length_bounds) == 2:
+        min_len = length_bounds[0]
+        max_len = length_bounds[1]
+    elif len(length_bounds) == 1:
+        max_len = length_bounds[0]
+        min_len = 0
+    else:
+        max_len = 2**32
+        min_len = 0
+    if ((len(read) < max_len) and (len(read) > min_len)):
+        return "Pass"
+    else:
+        return "Fail"
+			
+## This is a horrifying main function that takes all the above and uses it to filter your fastq file
+def main(input_fastq, output_file_prefix, gc_bounds = [0, 100], length_bounds = [0, 2**32], quality_threshold = 0, save_filtered = 'False'):
+    with open("test.fastq", "r") as fastq:
+        lines = fastq.readlines()
+        names=[item[:-1] for item in lines[::4]] 
+        reads=[item[:-1] for item in lines[1::4]]
```

Comment:
Обрезание последнего символа - плохой способ избавляться от переноса стоки по двум причинам:

В самой последней строке может не быть переноса (она же последняя), поэтому получится так, что мы убираем значимый символ и из-за этого в данном задании неправильно расчитывается среднее качество, а если в последнем риде была бы всего одна буква (маловероятно, но всё же), то вообще была бы ошибка ZeroDivisionError.
На некоторых системах (например, Windows) символами переноса строки являются \r\n, поэтому в данной ситуации наоборот останется один лишний символ

А ещё нужны пробелы
```suggestion
        names = [item.strip() for item in lines[::4]] 
        reads = [item.strip() for item in lines[1::4]]
```